tests/Aspire.Hosting.Tests/ContainerResourceTests.cs

```diff
@@ -45,4 +47,61 @@ public void AddContainerAddsAnnotationMetadataWithTag()
         Assert.Equal("none", containerAnnotation.Image);
         Assert.Null(containerAnnotation.Registry);
     }
+
+    [Fact]
+    public async Task AddContainerWithArgs()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+
+        var testResource = new TestResource("test", "connectionString");
+
+        var c1 = appBuilder.AddContainer("c1", "image2")
+            .WithEndpoint("ep", e =>
+            {
+                e.UriScheme = "http";
+                e.AllocatedEndpoint = new(e, "localhost", 1234);
+            });
+
+        var c2 = appBuilder.AddContainer("container", "none")
+             .WithArgs(context =>
+             {
+                 context.Args.Add("arg1");
+                 context.Args.Add(c1.GetEndpoint("ep"));
+                 context.Args.Add(testResource);
+             });
+
+        using var app = appBuilder.Build();
+
+        var args = await ArgumentEvaluator.GetArgumentListAsync(c2.Resource);
+
+        Assert.Collection(args,
+            arg => Assert.Equal("arg1", arg),
+            arg => Assert.Equal("http://localhost:1234", arg),
+            arg => Assert.Equal("connectionString", arg));
+
+        var manifest = await ManifestUtils.GetManifest(c2.Resource);
+
+        var expectedManifest =
```

Comment:
@vhvb1989 in case azd doesn' support expressions in the argument list

Reply:
Will take a look.
Thanks for the heads up.